pilot/pkg/serviceregistry/kube/controller/multicluster.go

```diff
@@ -176,7 +242,28 @@ func (m *Multicluster) DeleteMemberCluster(clusterID string) error {
 	if m.XDSUpdater != nil {
 		m.XDSUpdater.ConfigUpdate(&model.PushRequest{Full: true})
 	}
+	delete(m.remoteConfigController, clusterID)
```

Comment:
deleting it does not stop the controllers does it? I think we will leak this here and continue watching?

Reply:
delete will stop the remoteConfigController since the channel used when run remoteConfigController is closed:
close(m.remoteKubeControllers[clusterID].stopCh)